### PR TITLE
Define depthWriteEnabled as WGPUOptionalBool

### PR DESCRIPTION
**Pull request by beaufortfrancois**
FIX https://github.com/webgpu-native/webgpu-headers/issues/235

### CODE REVIEW

**Comment by beaufortfrancois** [comment]
@kainino0x Can you review?

File: webgpu.h [hunk 1]
```diff
@@ -407,6 +407,13 @@ typedef enum WGPUMipmapFilterMode {
     WGPUMipmapFilterMode_Force32 = 0x7FFFFFFF
 } WGPUMipmapFilterMode WGPU_ENUM_ATTRIBUTE;
 
+typedef enum WGPUOptionalBool {
+    WGPUOptionalBool_Undefined = 0x00000000,
+    WGPUOptionalBool_False = 0x00000001,
+    WGPUOptionalBool_True = 0x00000002,
```
Comment (kainino0x) [inline webgpu.h:413]: Should be False=0, True=1, Undefined=2
per https://github.com/webgpu-native/webgpu-headers/issues/235#issuecomment-1791098185

Comment (beaufortfrancois) [reply webgpu.h:413]: As raised in https://issues.chromium.org/issues/42241167#comment6, I've noticed Dawn asserts that "undefined" values must be 0 in `dawn_json_generator.py`. Shall we loosen this restriction in https://source.chromium.org/chromium/chromium/src/+/main:third_party/dawn/generator/dawn_json_generator.py;l=147;drc=83d3e7fc97d66a79c944c24b7ad18f549a5a150f?

Comment (kainino0x) [reply webgpu.h:413]: Oh yes, sorry I forgot to reply to that.
There's a strong reason to have False=0, True=1 so yes, I think we need to do that.

Comment (beaufortfrancois) [reply webgpu.h]: @kainino0x I've updated this PR with your suggested changes.

**Review by kainino0x — APPROVED** [review]
LGTM!

**Pull request merged**